### PR TITLE
LOD body fixes

### PR DESCRIPTION
**Pull request by Schneegans**
This adds the following:
* `"url": "offline"` can now be used to prevent a `TileSourceWebMapService` from accessing a map server. Only cached data will be used.
* The terrain shader works with OpenGL 3.3.
* If loading of a cached tile fails, it will be removed now effectively triggering a re-download.
* `csp-lod-bodies` will now only print errors for critical events. If downloading a tile failed, this not considered critical.
* The slope-range and height-range slider now work as supposed.

### CODE REVIEW

**Comment by JonasGilg** [comment]
On Ubuntu I get the following error:
```
[E] vista ............. ###ERROR### OpenGL error 1281 ( invalid value ) in file /localdata1/ClionProjects/cosmoscout-vr/externals/vista/VistaCoreLibs/VistaOGLExt/VistaBufferObject.cpp in line 69 detected.
[E] vista ............. ###ERROR### [VistaGLSLShader] Error while linking shaders. See below for details.
[E] vista .............     [VistaGLSLShader] Program info log:
[I] vista ............. error: linking with uncompiled/unspecialized shader
[I] vista ............. 
[E] vista .............     [VistaGLSLShader] Vertex shader #0 info log: 
[I] vista ............. 
[E] vista .............     [VistaGLSLShader] Fragment shader #0 info log: 
[I] vista ............. 0:128(2): error: `return' with wrong type int, in function `VP_getShadow' returning float
[I] vista ............. 0:139(20): error: sampler arrays indexed with non-constant expressions are forbidden in GLSL 1.30 and later
[I] vista ............. 
[W] vista ............. **WARNING** Shader program with id 257 is not ready for use. Bind has no effect.
[E] vista ............. ###ERROR### OpenGL error 1282 ( invalid operation ) in file /localdata1/ClionProjects/cosmoscout-vr/externals/vista/VistaCoreLibs/VistaOGLExt/VistaBufferObject.cpp in line 69 detected.
```

The LoD-Body doesn't render.

**Comment by Schneegans** [comment]
Weird. This should have been fixed by https://github.com/cosmoscout/cosmoscout-vr/pull/195/commits/b21e8a8d13f2efc7b56dff4c97a03ebf5a995673...? Are you sure you're testing the correct branch?

**Comment by JonasGilg** [comment]
It kinda works now. I still get the following two lines:
```
[E] vista ............. ###ERROR### OpenGL error 1281 ( invalid value ) in file /localdata1/ClionProjects/cosmoscout-vr/externals/vista/VistaCoreLibs/VistaOGLExt/VistaBufferObject.cpp in line 69 detected.
[E] vista ............. ###ERROR### OpenGL error 1282 ( invalid operation ) in file /localdata1/ClionProjects/cosmoscout-vr/externals/vista/VistaCoreLibs/VistaOGLExt/VistaBufferObject.cpp in line 69 detected.
```

And when disabling the atmospheres in the gui the lod-body stops rendering too o.O.

File: plugins/csp-lod-bodies/src/TileSourceWebMapService.cpp [hunk 1]
```diff
@@ -42,24 +42,40 @@ enum class CopyPixels { eAll, eAboveDiagonal, eBelowDiagonal };
...
   try {
     cacheFile = source->loadData(level, x, y);
   } catch (std::exception const& e) {
-    logger().error("Tile loading failed: {}", e.what());
+    // This is not critical, the planet will just not refine any further.
+    logger().debug("Tile loading failed: {}", e.what());
     return false;
   }
 
+  // Data is not available. That's most likely do to our server being offline.
```
Comment (JonasGilg) [inline plugins/csp-lod-bodies/src/TileSourceWebMapService.cpp:58]:
```suggestion
  // Data is not available. That's most likely due to our server being offline.
```

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**